zilencer/views.py

```diff
@@ -301,13 +330,14 @@ def unregister_remote_push_device(
     return json_success(request)
 
 
-@has_request_variables
+@typed_endpoint
 def unregister_all_remote_push_devices(
     request: HttpRequest,
     server: RemoteZulipServer,
-    user_id: Optional[int] = REQ(json_validator=check_int, default=None),
-    user_uuid: Optional[str] = REQ(default=None),
-    realm_uuid: Optional[str] = REQ(default=None),
+    *,
+    user_id: Optional[Json[int]] = None,
+    user_uuid: Optional[str] = None,
+    realm_uuid: Optional[str] = None,
```

Comment:
Should these be marked as the UUID types (https://docs.pydantic.dev/2.0/usage/types/uuids/)?

Reply:
Yes, that makes a lot of sense, I would have to look into validating the uuid is being send as str or uuid object itself.

Reply:
Implemented it only to realise the type will be string and not UUID, [Fast Api ref](https://fastapi.tiangolo.com/tutorial/extra-data-types/#other-data-types).

Reply:
That FastAPI reference seems to just be saying that if you have a `UUID4` object in Python, that FastAPI will represent it over the wire as a string, which is of course what one wants, but the question here is whether we should turn it into a `UUID` object with automatic validation in Python here.